### PR TITLE
Refactor the cluster package

### PR DESCRIPTION
**Pull request by ejj**
This patch is almost entirely aesthetic and interface changes.  Doesn't need to be reviewed all that carefully, should just be skimmed.

### CODE REVIEW

File: cluster/google/google.go [hunk 1]
```diff
@@ -61,49 +67,46 @@ type gceCluster struct {
...
 	}
 
-	clst.projID = "declarative-infrastructure"
-	clst.ns = namespace
-	clst.imgURL = fmt.Sprintf(
-		"%s/%s",
-		computeBaseURL,
-		"ubuntu-os-cloud/global/images/ubuntu-1604-xenial-v20160921")
+	clst := Cluster{
+		projID:    "declarative-infrastructure",
```
Comment (mmussomele) [inline cluster/google/google.go]: quilt?

File: cluster/google/google.go [hunk 2]
```diff
@@ -61,49 +67,46 @@ type gceCluster struct {
 	id int    // the id of the cluster, used externally
 }
 
-// Create a GCE cluster.
+// New creates a GCE cluster.
 //
 // Clusters are differentiated (namespace) by setting the description and
 // filtering off of that.
-//
-// XXX: A lot of the fields are hardcoded.
```
Comment (mmussomele) [inline cluster/google/google.go]: Does this block? If so, might want to note in the comment.

**Comment by mmussomele** [comment]
Looks good, just two small comments

File: cluster/google/google.go (hunk 2 above)
Comment (ejj) [inline cluster/google/google.go]: So both of these comments are sort of general improvements that need to be made to the gce provider.  I think we separately need to do a pass over it and clean things up, but I don't really thing this is the right patch for it.

**Pull request merged**